### PR TITLE
feat(chat-e2e): shared prompt folder icon tests

### PR DESCRIPTION
**Pull request by irinakartun**
**Description:**

implemented tests for shared prompt folder icons

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [ ] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1954.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1954.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1954.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1954.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1954.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat-e2e/src/assertions/folderAssertion.ts [hunk 1]
```diff
@@ -364,4 +364,51 @@ export class FolderAssertion {
...
+    expectedState === 'visible'
+      ? await expect
+          .soft(arrowIcon, ExpectedMessages.sharedEntityIconIsVisible)
+          .toBeVisible()
+      : await expect
+          .soft(arrowIcon, ExpectedMessages.sharedEntityIconIsNotVisible)
+          .toBeHidden();
+  }
+
+  public async assertFolderArrowIconColor(
```
Comment (nartovm) [inline apps/chat-e2e/src/assertions/folderAssertion.ts]: Are we talking here about shared folder arrow only or about any arrow? Perhaps you should name this method assertSharedFolderArrowIconColor?

File: apps/chat-e2e/src/testData/expectedConstants.ts [hunk 2]
```diff
@@ -64,6 +64,7 @@ export const ExpectedConstants = {
     'Model changing will stop sharing and other users will no longer see this conversation.',
   renameSharedPromptMessage:
     'Renaming will stop sharing and other users will no longer see this prompt.',
+  unshareFolderMessage: 'Are you sure that you want to unshare this folder?',
```
Comment (nartovm) [inline apps/chat-e2e/src/testData/expectedConstants.ts:67]: Shouldn't we have 2 separate constants for conversations and prompts?

File: apps/chat-e2e/src/tests/parametrizedReplay.test.ts [hunk 3]
```diff
@@ -285,7 +285,7 @@ dialTest(
           chatBar.importButton.click(),
         );
         await conversations.openEntityDropdownMenu(conversation.name);
-        await conversations.selectEntityMenuOption(MenuOptions.replay, {
+        await conversationDropdownMenu.selectMenuOption(MenuOptions.replay, {
```
Comment (nartovm) [inline apps/chat-e2e/src/tests/parametrizedReplay.test.ts:288]: Why do we have a separate method for selectDuplicateMenuOption, but do not use the same approach here?

File: apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts [hunk 4]
```diff
@@ -213,7 +213,7 @@ dialTest(
 
     await dialTest.step('Click on Duplicate', async () => {
       await conversations.openEntityDropdownMenu(firstConversation.name, 2);
-      await conversationDropdownMenu.selectMenuOption(MenuOptions.duplicate);
+      await conversations.selectDuplicateMenuOption();
```
Comment (nartovm) [inline apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts]: We should indicate that we are waiting for the request result inside of the method and that is not 100% equal to await conversations.selectEntityMenuOption(MenuOptions.duplicate)

File: apps/chat-e2e/src/tests/selectPromptPanelEntity.test.ts [hunk 5]
```diff
@@ -1435,7 +1440,7 @@ dialTest(
         await prompts.openEntityDropdownMenu(singlePrompt.name);
         await promptDropdownMenu.selectMenuOption(MenuOptions.select);
         await promptFilter.openFilterDropdownMenu();
-        await promptFilterDropdownMenu.selectMenuOption(
+        await promptFilterDropdownMenu.selectOption(
```
Comment (nartovm) [inline apps/chat-e2e/src/tests/selectPromptPanelEntity.test.ts]: why renaming if we literally use Filter**Menu**Options inside and other methods and fields like getAllMenuOptions, abstract menuOptions() that contain the word 'menu"?

File: apps/chat-e2e/src/tests/sharedPromptFolderIcons.test.ts [hunk 6]
```diff
@@ -0,0 +1,453 @@
...
+    let shareResponse: ShareByLinkResponseModel;
+    const newFolderName = GeneratorUtil.randomString(5);
+
+    await dialTest.step(
+      'Prepare nested folders hierarchy and move prompt lowest level folder',
+      async () => {
+        nestedFolders = promptData.prepareNestedFolder(nestedLevels);
+        const lowLevelFolderPrompt = promptData.preparePromptInFolder(
+          GeneratorUtil.randomString(5),
+          GeneratorUtil.randomString(5),
```
Comment (nartovm) [inline apps/chat-e2e/src/tests/sharedPromptFolderIcons.test.ts]: Why use random names if we could use something that makes sence and much easier to track and debug, such are names from ExpectedConstants + counter?

File: apps/chat-e2e/src/assertions/folderAssertion.ts (hunk 1 above)
Comment (irinakartun) [reply apps/chat-e2e/src/assertions/folderAssertion.ts]: agree, done

File: apps/chat-e2e/src/testData/expectedConstants.ts (hunk 2 above)
Comment (irinakartun) [reply apps/chat-e2e/src/testData/expectedConstants.ts:67]: no, here we are talking about shared folder and messages are identical in this case

File: apps/chat-e2e/src/tests/parametrizedReplay.test.ts (hunk 3 above)
Comment (irinakartun) [reply apps/chat-e2e/src/tests/parametrizedReplay.test.ts:288]: 'Duplicate' option is available for both side-bar entities - conversation and prompt, that's why I created a common method selectDuplicateMenuOption() in SideBarEntities class. However, 'Replay', 'Playback'etc options are defined only for conversations.

File: apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts (hunk 4 above)
Comment (irinakartun) [reply apps/chat-e2e/src/tests/chatSelectionFunctionality.test.ts]: The .selectEntityMenuOption() method no longer exists

File: apps/chat-e2e/src/tests/selectPromptPanelEntity.test.ts (hunk 5 above)
Comment (irinakartun) [reply apps/chat-e2e/src/tests/selectPromptPanelEntity.test.ts]: thanks, fixed

File: apps/chat-e2e/src/tests/sharedPromptFolderIcons.test.ts (hunk 6 above)
Comment (irinakartun) [reply apps/chat-e2e/src/tests/sharedPromptFolderIcons.test.ts]: utilizing random data in software testing is considered a good practice. This approach reflects the variety of input that can occur in a real-world scenario. Given that the entity name is unique, using ExpectedConstants+counter could cause confusion and test flakiness if an entity with the same name already exists in the app

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1954.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1954.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat-e2e/src/tests/parametrizedReplay.test.ts (hunk 3 above)
Comment (irinakartun) [reply apps/chat-e2e/src/tests/parametrizedReplay.test.ts:288]: removed selectDuplicateMenuOption and replaced it with selectMenuOption() call

File: apps/chat-e2e/src/tests/sharedPromptFolderIcons.test.ts (hunk 6 above)
Comment (irinakartun) [reply apps/chat-e2e/src/tests/sharedPromptFolderIcons.test.ts]: replaced with meaningful names

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1954.nightly-test.deltixhub.io
>E2E tests status: success

**Review by nartovm — APPROVED** [review]

**Comment by irinakartun** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1954.nightly-test.deltixhub.io
>E2E tests status: success

**Pull request merged**